stdlib/public/Concurrency/Actor.cpp

```diff
@@ -888,29 +847,65 @@ static_assert(sizeof(ActiveActorStatus) == ACTIVE_ACTOR_STATUS_SIZE,
 /// exist yet. As a result, the subset of rules that currently apply
 /// are (1), (3), (5), (6).
 class DefaultActorImpl : public HeapObject {
+  struct Header {
 #if SWIFT_CONCURRENCY_ACTORS_AS_LOCKS
-  // If actors are locks, we don't need to maintain any extra bookkeeping in the
-  // ActiveActorStatus since all threads which are contending will block
-  // synchronously, no job queue is needed and the lock will handle all priority
-  // escalation logic
-  Mutex drainLock;
+    // If actors are locks, we don't need to maintain any extra bookkeeping in the
+    // ActiveActorStatus since all threads which are contending will block
+    // synchronously, no job queue is needed and the lock will handle all priority
+    // escalation logic
+    Mutex drainLock;
 #else
-  // Note: There is some padding that is added here by the compiler in order to
-  // enforce alignment. This is space that is available for us to use in
-  // the future
-  alignas(sizeof(ActiveActorStatus)) char StatusStorage[sizeof(ActiveActorStatus)];
+    // Note: There is some padding that is added here by the compiler in order to
+    // enforce alignment. This is space that is available for us to use in
+    // the future
+    alignas(sizeof(ActiveActorStatus)) char StatusStorage[sizeof(ActiveActorStatus)];
+#endif
+    // TODO (rokhinip): Make this a flagset
+    bool isDistributedRemoteActor;
+  } header;
+
+#if !SWIFT_CONCURRENCY_ACTORS_AS_LOCKS
+  using SimpleQueue = swift::SimpleQueue<Job *, JobQueueTraits>;
+  using PriorityQueue = swift::PriorityQueue<Job *, JobQueueTraits>;
+
+  struct Footer {
+    // When enqueued, jobs are atomically added to a linked list with the head
+    // stored inside ActiveActorStatus. This list contains jobs in the LIFO order
+    // regardless of their priorities.
+    //
+    // After that processing thread collects them and adds to the
+    // `prioritisedJobs` while holding the actor lock.
+    //
+    PriorityQueue prioritisedJobs;
+  };
+
+  struct Padding {
+    enum {
+      // Available space
+      x1 = (int)(sizeof(void *) * NumWords_DefaultActor),
+      // ... subtracting size of footer
+      x2 = x1 - (int)sizeof(Footer),
+      // ... ensuring footer is aligned
+      x3 = x2 - (int)(x2 % alignof(Footer)),
```

Comment:
This will not work correctly if `alignof(Footer)` is larger then `sizeof(HeapObject)`. To compute everything correctly, `sizeof(HeapObject)` should be added to the `x1` and subtracted later.